app/partials/app.cjsx

```diff
@@ -3,15 +3,81 @@ auth = require 'panoptes-client/lib/auth'
 IOStatus = require './io-status'
 MainHeader = require './main-header'
 MainFooter = require './main-footer'
+GeordiClient = require 'zooniverse-geordi-client'
 {generateSessionID} = require '../lib/session'
 
+class GeordiLogger
+  constructor: (@state, @geordi) ->
+
+  keys: {}
+
+  geordi: null
+
+  instance: =>
+    @geordi = @geordi || @makeGeordi @keys?.projectToken
+
+  makeGeordi: (projectSlug) ->
+    new GeordiClient
+      server: @state?.env
+      projectToken: projectSlug || ''
+      zooUserIDGetter: () => @state.user?.id
+      subjectGetter: () => @keys?.subjectID
+
+  makeHandler: (defType) ->
+    instance = @instance
+    (eventData, eventType) ->
+        eventType = defType if typeof linkType isnt 'string'
+        instance()?.logEvent
+          type: eventType
+          data: "\"#{eventData}\""
+
+  logEvent: (logEntry) ->
+    geordi = @instance()
+    newEntry = Object.assign {}, logEntry, @keys
+    geordi?.logEvent newEntry
+    console.log 'No logger available for event ', JSON.stringify(logEntry) unless geordi?.logEvent
+
+  remember: (keyVals) ->
+    rebuild = keyVals?.projectToken? && (keyVals?.projectToken != @geordi?.projectToken)
+    @geordi = @makeGeordi keyVals.projectToken if rebuild
+    @keys = Object.assign {}, @keys, keyVals
+
+  forget: (forgetKeys) ->
+    rebuild = false
+    for key in forgetKeys
+      rebuild = true if key == 'projectToken'
+      delete @keys[key]
+
+    @geordi = @makeGeordi '' if rebuild
+
 module.exports = React.createClass
   displayName: 'PanoptesApp'
 
+  geordiLogger: null
```

Comment:
An interface that allows a Panoptes to impersonate an Ouroboros project to the Geordi client?

It's essentially a wrapper so that `zooUserIDGetter` and `subjectGetter` work in the client.